src/Symfony/Component/HttpClient/Response/AsyncContext.php

```diff
@@ -151,6 +151,11 @@ public function getResponse(): ResponseInterface
     public function replaceRequest(string $method, string $url, array $options = []): ResponseInterface
     {
         $this->info['previous_info'][] = $this->response->getInfo();
+        if (null !== $onProgress = $options['on_progress'] ?? null) {
+            $options['on_progress'] = function (int $dlNow, int $dlSize, array $info) use ($onProgress) {
+                $onProgress($dlNow, $dlSize, $info + $this->info);
```

Comment:
it should be `$this->info + $info` as in `AsyncResponse::getInfo()`
I think this non-static closure might create a circular ref. We should use a reference to &$this->info to prevent it.

Reply:
fixed